From c998d0c12e83b77b4320fb4f3bef791a840dbec8 Mon Sep 17 00:00:00 2001
From: Ashish <apaliwal@apache.org>
Date: Thu, 6 Aug 2015 13:55:43 -0700
Subject: [PATCH 14/37] FLUME-2753 Error when specifying empty replace string
 in Search and Replace Interceptor

(Johny Rufus via Ashish Paliwal)
---
 .../interceptor/SearchAndReplaceInterceptor.java   |    8 +++++---
 .../TestSearchAndReplaceInterceptor.java           |   11 +++++++++++
 2 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/flume-ng-core/src/main/java/org/apache/flume/interceptor/SearchAndReplaceInterceptor.java b/flume-ng-core/src/main/java/org/apache/flume/interceptor/SearchAndReplaceInterceptor.java
index 6f5c146..b8588cd 100644
--- a/flume-ng-core/src/main/java/org/apache/flume/interceptor/SearchAndReplaceInterceptor.java
+++ b/flume-ng-core/src/main/java/org/apache/flume/interceptor/SearchAndReplaceInterceptor.java
@@ -105,9 +105,11 @@ public class SearchAndReplaceInterceptor implements Interceptor {
           " (may not be empty)");
 
       replaceString = context.getString(REPLACE_STRING_KEY);
-      Preconditions.checkNotNull(replaceString,
-          "Must supply a replacement string " + REPLACE_STRING_KEY +
-          " (empty is ok)");
+      // Empty replacement String value or if the property itself is not present
+      // assign empty string as replacement
+      if(replaceString == null) {
+        replaceString = "";
+      }
 
       searchRegex = Pattern.compile(searchPattern);
 
diff --git a/flume-ng-core/src/test/java/org/apache/flume/interceptor/TestSearchAndReplaceInterceptor.java b/flume-ng-core/src/test/java/org/apache/flume/interceptor/TestSearchAndReplaceInterceptor.java
index b39f912..2ab15f5 100644
--- a/flume-ng-core/src/test/java/org/apache/flume/interceptor/TestSearchAndReplaceInterceptor.java
+++ b/flume-ng-core/src/test/java/org/apache/flume/interceptor/TestSearchAndReplaceInterceptor.java
@@ -84,4 +84,15 @@ public class TestSearchAndReplaceInterceptor {
     context.put("replaceString", "REDACTED");
     testSearchReplace(context, input, output);
   }
+
+  @Test
+  public void testReplaceEmpty() throws Exception {
+    final String input = "Abc123@test.com";
+    final String output = "@test.com";
+    Context context = new Context();
+    context.put("searchPattern", "^[A-Za-z0-9_]+");
+    testSearchReplace(context, input, output);
+    context.put("replaceString", "");
+    testSearchReplace(context, input, output);
+  }
 }
\ No newline at end of file
-- 
1.7.9.5

